python_modules/libraries/dagster-duckdb/dagster_duckdb/io_manager.py

```diff
@@ -107,6 +110,26 @@ def duckdb_io_manager(init_context):
     return duckdb_io_manager
 
 
+def build_configurable_duckdb_io_manager(
+    type_handlers: Sequence[DbTypeHandler], default_load_type: Optional[Type] = None
+):
+    class ConfigurableDuckDBIOManager(ConfigurableIOManagerFactory):
+        database: str
+        schema_: Optional[str] = None  # schema is a reserved word for pydantic
```

Comment:
i tried that and it doesn't seem to work as expected. this is the type-ahead i get 
<img width="1025" alt="Screenshot 2023-03-09 at 3 38 54 PM" src="https://user-images.githubusercontent.com/19783112/224151743-90c2e7d7-e29b-40ba-a697-5ada77a0c9de.png">

Reply:
I think we want do when @benpankow said but I am also seeing some unexpected behavior.


For example I have the code:

```py
from dagster import Config, asset, Resource
from dagster._config.structured_config import infer_schema_from_config_class
from pydantic import Field


class ConfigWithSchema(Config):
    schema_: str = Field(
        alias="schema",
        description="A string",
    )


config_object = ConfigWithSchema(schema="foo")
print(config_object.dict())
print(config_object._as_config_dict())  # same behavior as .dict()
print(config_object.dict(by_alias=True))
print(config_object.schema_)
print(infer_schema_from_config_class(ConfigWithSchema).config_type.fields)


@asset
def an_asset(config: ConfigWithSchema):
    print(config.schema_)
```

Running this prints out

```
{'schema_': 'foo'}
{'schema_': 'foo'}
{'schema': 'foo'}
foo
{'schema': Field(<dagster._config.source.StringSourceType object at 0x10e45ace0>, default=@, is_required=True)}
```

Which I *think* should be right. Good question if `_as_config_dict()` should respect aliases or not. Seems like it should since it generates a config schema with `schema`.

The config editor also shows `schema`. 

![Screen Shot 2023-03-09 at 4 58 00 PM](https://user-images.githubusercontent.com/28738937/224170801-b9c1ab09-c4a8-44d7-83a5-d43cd9c6a088.png)

I think this is fairly important as users will copy paste. @jamiedemaria it looks like you are seeing different behavior here. Is it different for Resource classes and Config classes?

Reply:
When I set up my io manager like this
```
resources = {
    "io_manager": ConfigurableDuckDBPandasIOManager(database="my_db.duckdb", schema="my_schema")
}
```
and materialize an asset, i also get this error
```
dagster._core.errors.DagsterSubprocessError: During multiprocess execution errors occurred in child processes:
In process 16033: dagster._core.errors.DagsterInvalidConfigError: Incorrect values passed to .configured
Error 1: Received unexpected config entry "schema_" at the root. Expected: "{ database: (String | { env: String }) schema?: (String | { env: String }) }".

Stack Trace:
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_core/executor/child_process_executor.py", line 79, in _execute_command_in_child_process
for step_event in command.execute():
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_core/executor/multiprocess.py", line 93, in execute
yield from execute_plan_iterator(
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_core/execution/api.py", line 1094, in __iter__
yield from self.execution_context_manager.prepare_context()
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_utils/__init__.py", line 536, in generate_setup_events
obj = next(self.generator)
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_core/execution/context_creation_pipeline.py", line 261, in execution_context_event_generator
yield from resources_manager.generate_setup_events()
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_utils/__init__.py", line 536, in generate_setup_events
obj = next(self.generator)
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_core/execution/resources_init.py", line 260, in resource_initialization_event_generator
yield from _core_resource_initialization_event_generator(
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_core/execution/resources_init.py", line 189, in _core_resource_initialization_event_generator
for event in manager.generate_setup_events():
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_utils/__init__.py", line 536, in generate_setup_events
obj = next(self.generator)
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_core/execution/resources_init.py", line 328, in single_resource_event_generator
resource_def.resource_fn(context)
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_config/structured_config/__init__.py", line 340, in setup_context_resources_and_call
return self._resource.resource_fn(context_with_key_mapping)
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_config/structured_config/__init__.py", line 503, in initialize_and_run
with_nested_resources = self._resolve_and_update_nested_resources(context)
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_config/structured_config/__init__.py", line 500, in _resolve_and_update_nested_resources
return self._with_updated_values(to_update)
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_config/structured_config/__init__.py", line 447, in _with_updated_values
return self.__class__(**{**self._as_config_dict(), **values})
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_annotations.py", line 121, in __init__
undecorated_init(self, *args, **kwargs)
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_config/structured_config/__init__.py", line 646, in __init__
ConfigurableResource.__init__(self, **data)
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_annotations.py", line 121, in __init__
undecorated_init(self, *args, **kwargs)
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_config/structured_config/__init__.py", line 412, in __init__
curried_schema = _curry_config_schema(schema, resolved_config_dict)
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_config/structured_config/__init__.py", line 210, in _curry_config_schema
return DefinitionConfigSchema(_apply_defaults_to_schema_field(schema_field, data))
File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_config/structured_config/__init__.py", line 158, in _apply_defaults_to_schema_field
raise DagsterInvalidConfigError(


  File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_core/execution/api.py", line 988, in pipeline_execution_iterator
    for event in pipeline_context.executor.execute(pipeline_context, execution_plan):
  File "/Users/jamie/dev/dagster/python_modules/dagster/dagster/_core/executor/multiprocess.py", line 306, in execute
    raise DagsterSubprocessError(
```